mir_eval/sonify.py

```diff
@@ -147,34 +154,42 @@ def __interpolator(x):
             return value
         return __interpolator
 
+    # print(f"sonify.time_frequency length {length}, n_times {n_times}, gram shape {gram.shape}, times shape {times.shape}")
+
     # Threshold the tfgram to remove non-positive values
     gram = np.maximum(gram, 0)
 
     # Pre-allocate output signal
     output = np.zeros(length)
     time_centers = np.mean(times, axis=1) * float(fs)
 
+    # Not really a true spectral energy, just a summation for optimisation.
+    spectral_sums = np.sum(gram, axis = 1)
     for n, frequency in enumerate(frequencies):
+        if spectral_sums[n] < 0.01: # TODO set threshold intelligently.
```

Comment:
If we're going to have a threshold, it should probably be either parametrized or relative to the values in `gram`